File: DoorsPOS.WinApp/CS/frmReturns.cs
// VBConversions Note: VB project level imports
using System.Data;
using System.Collections.Generic;
using PDSA.Data;
using PDSA.WinForms.Forms;
using PDSA.Security;
using System.Diagnostics;
using PDSA.Common;
using PDSA.WinForms;
using PDSA.WinForms.Security;
using PDSA.PDSACryptography;
using DoorsPOS.Utility;
using Microsoft.VisualBasic;
using System.Collections;
using System;
using DoorsPOS.BusinessLayer;
using System.Windows.Forms;
using System.Drawing;
using System.Linq;
using DoorsPOS.DataLayer;
// End of VB project level imports

using PDSA.DataLayer.DataClasses;
using PDSA.Validation;
using DevExpress.XtraEditors;

namespace DoorsPOS.WinApp
{
	public partial class frmReturns
	{
		public frmReturns()
		{
			InitializeComponent();
		}
		
		stocksManager mgr8;
		public void frmReturns_FormClosed(object sender, FormClosedEventArgs e)
		{
			this.Dispose();
		}
		
		public void frmReturns_FormClosing(object sender, FormClosingEventArgs e)
		{
			if (btnSave.Enabled == true)
			{
				e.Cancel = true;
				MessageBox.Show("Save or Cancel the Entry First Before Closing the Form.");
			}
		}
		
		public void frmReturns_Load(object sender, EventArgs e)
		{
			this.Cursor = Cursors.WaitCursor;
			LoadMember();
			this.Cursor = Cursors.Default;
		}
		
		private void LoadMember()
		{
			membersManager mgrs = new membersManager();
			mgrs.DataObject.OrderByFilter = membersData.OrderByFilters.lastname;
			membersCollection cols = default(membersCollection);
			cols = mgrs.BuildCollection();
			leSupplier.Properties.DisplayMember = "lastname";
			leSupplier.Properties.ValueMember = "CustID";
			leSupplier.Properties.DataSource = cols;
		}
		
		public void btnItemSearch_Click(object sender, EventArgs e)
		{
			frmItemSearch frmitemserch = new frmItemSearch();
			frmitemserch.ShowDialog();
			txtItem.Text = string.Empty;
			txtStckId.Text = string.Empty;
			txtItem.Text = frmitemserch.vItem.ToString();
			txtStckId.Text = frmitemserch.vStckid.ToString();
			if (txtStckId.Text == string.Empty)
			{
				return;
			}
			else
			{
				ItemsearchExecute();
			}
			
		}
		
		private void ItemsearchExecute()
		{
			stocksCollection col2;
			int nStckid = 0;
			try
			{
				mgr8 = new stocksManager();
				mgr8.DataObject.WhereFilter = stocksData.WhereFilters.PrimaryKey;
				mgr8.DataObject.Entity.stckid = int.Parse(txtStckId.Text);
				col2 = mgr8.BuildCollection();
				if (mgr8.DataObject.RowsAffected > 0)
				{
					nStckid = mgr8.DataObject.Entity.stckid;
					txtPrice.Text = Strings.FormatNumber((mgr8.DataObject.Entity.cost).ToString(), 2, (Microsoft.VisualBasic.TriState) Microsoft.VisualBasic.TriState.UseDefault, (Microsoft.VisualBasic.TriState) Microsoft.VisualBasic.TriState.UseDefault, (Microsoft.VisualBasic.TriState) Microsoft.VisualBasic.TriState.UseDefault);
					ceQty.Focus();
				}
				
			}
			catch (PDSAValidationException ex)
			{
				MessageBox.Show(ex.Message);
			}
			catch (Exception ex)
			{
				MessageBox.Show(ex.ToString());
			}
		}
		
		public void btnClose_Click(object sender, EventArgs e)
		{
			this.Close();
		}
		
		public void btnSave_Click(object sender, EventArgs e)
		{
			if (txtItem.Text == string.Empty)
			{
				MessageBox.Show("Please select an item to be returned.");
				txtItem.Focus();
				return;
			}
			
			if (ceQty.Value <= 0)
			{
				MessageBox.Show("Please enter the number of items to be returned.");
				ceQty.Focus();
				return;
			}
			
			if (deBoDate.Text == string.Empty)
			{
				MessageBox.Show("Please enter the date the item was returned.");
				deBoDate.Focus();
				return;
			}
			
			if (leSupplier.Text == string.Empty)
			{
				MessageBox.Show("Please select the supplier of the item to be returned.");
				leSupplier.Focus();
				return;
			}
			
			
			PDSATransaction TranBo = new PDSATransaction();
			boManager mgrbo = new boManager();
			
			try
			{
				mgrbo.DataObject.TransactionType = PDSATransactionType.Insert;
				mgrbo.DataObject.Entity.amnt = 0;
				mgrbo.DataObject.Entity.bodate = DateTime.Parse(deBoDate.Text);
				mgrbo.DataObject.Entity.prevavlbl = 0;
				mgrbo.DataObject.Entity.price = 0;
				mgrbo.DataObject.Entity.qty = (int) ceQty.Value;
				mgrbo.DataObject.Entity.sInsertid = System.Convert.ToString(PDSAAppConfig.CurrentLoginID);
				mgrbo.DataObject.Entity.sLastUpdateid = System.Convert.ToString(PDSAAppConfig.CurrentLoginID);
				mgrbo.DataObject.Entity.stckid = int.Parse(txtStckId.Text.Trim());
				mgrbo.DataObject.Entity.supcode = System.Convert.ToInt32(leSupplier.EditValue);
				TranBo.Add(mgrbo.DataObject);
				
				spBackOrderManager mgrspBo = new spBackOrderManager();
				mgrspBo.Entity.StckIid = int.Parse(txtStckId.Text);
				TranBo.Add(mgrspBo.DataObject);
				
				TranBo.Execute();
				DisableControls();
				MessageBox.Show("Back Order Entry Successfully Saved", "Saved");
				InitControls();
			}
			catch (PDSAValidationException ex)
			{
				MessageBox.Show(ex.Message);
				return;
			}
			catch (Exception ex)
			{
				MessageBox.Show(ex.ToString());
				return;
			}
			
		}
		
		private void EnableControls()
		{
			btnSave.Enabled = true;
			btnCancel.Enabled = true;
			GroupControl1.Enabled = true;
			btnClose.Enabled = false;
			btnNew.Enabled = false;
			deBoDate.Focus();
		}
		
		private void DisableControls()
		{
			btnSave.Enabled = false;
			btnCancel.Enabled = false;
			GroupControl1.Enabled = false;
			btnClose.Enabled = true;
			btnNew.Enabled = true;
			btnNew.Focus();
		}
		
		private void InitControls()
		{
			deBoDate.EditValue = DateAndTime.Today;
			txtItem.Text = "";
			txtStckId.Text = "";
			txtPrice.Text = "";
			ceQty.Value = 0;
		}
		
		public void btnCancel_Click(object sender, EventArgs e)
		{
			DisableControls();
			InitControls();
		}
		
		public void btnNew_Click(object sender, EventArgs e)
		{
			EnableControls();
			InitControls();
		}
		
		public void ceQty_LostFocus(object sender, EventArgs e)
		{
			btnSave.Focus();
		}
	}
	
}

File: DoorsPOS.WinApp/frmReturns.vb
Imports PDSA.DataLayer.DataClasses
Imports PDSA.Validation
Imports DevExpress.XtraEditors
Public Class frmReturns

    Dim mgr8 As stocksManager
    Private Sub frmReturns_FormClosed(sender As Object, e As FormClosedEventArgs) Handles Me.FormClosed
        Me.Dispose()
    End Sub

    Private Sub frmReturns_FormClosing(sender As Object, e As FormClosingEventArgs) Handles Me.FormClosing
        If btnSave.Enabled = True Then
            e.Cancel = True
            MessageBox.Show("Save or Cancel the Entry First Before Closing the Form.")
        End If
    End Sub

    Private Sub frmReturns_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Me.Cursor = Cursors.WaitCursor
        LoadMember()
        Me.Cursor = Cursors.Default
    End Sub

    Private Sub LoadMember()
        Dim mgrs As membersManager = New membersManager()
        mgrs.DataObject.OrderByFilter = membersData.OrderByFilters.lastname
        Dim cols As membersCollection
        cols = mgrs.BuildCollection()
        leSupplier.Properties.DisplayMember = "lastname"
        leSupplier.Properties.ValueMember = "CustID"
        leSupplier.Properties.DataSource = cols
    End Sub

    Private Sub btnItemSearch_Click(sender As Object, e As EventArgs) Handles btnItemSearch.Click
        Dim frmitemserch As New frmItemSearch
        frmitemserch.ShowDialog()
        txtItem.Text = String.Empty
        txtStckId.Text = String.Empty
        txtItem.Text = frmitemserch.vItem.ToString
        txtStckId.Text = frmitemserch.vStckid.ToString
        If txtStckId.Text = String.Empty Then
            Exit Sub
        Else
            ItemsearchExecute()
        End If

    End Sub

    Private Sub ItemsearchExecute()
        Dim col2 As stocksCollection
        Dim nStckid As Integer = 0
        Try
            mgr8 = New stocksManager()
            mgr8.DataObject.WhereFilter = stocksData.WhereFilters.PrimaryKey
            mgr8.DataObject.Entity.stckid = CInt(txtStckId.Text)
            col2 = mgr8.BuildCollection()
            If mgr8.DataObject.RowsAffected > 0 Then
                nStckid = mgr8.DataObject.Entity.stckid
                txtPrice.Text = FormatNumber(CStr(mgr8.DataObject.Entity.cost), 2)
                ceQty.Focus()
            End If

        Catch ex As PDSAValidationException
            MessageBox.Show(ex.Message)
        Catch ex As Exception
            MessageBox.Show(ex.ToString())
        End Try
    End Sub

    Private Sub btnClose_Click(sender As Object, e As EventArgs) Handles btnClose.Click
        Me.Close()
    End Sub

    Private Sub btnSave_Click(sender As Object, e As EventArgs) Handles btnSave.Click
        If txtItem.Text = String.Empty Then
            MessageBox.Show("Please select an item to be returned.")
            txtItem.Focus()
            Exit Sub
        End If

        If ceQty.Value <= 0 Then
            MessageBox.Show("Please enter the number of items to be returned.")
            ceQty.Focus()
            Exit Sub
        End If

        If deBoDate.Text = String.Empty Then
            MessageBox.Show("Please enter the date the item was returned.")
            deBoDate.Focus()
            Exit Sub
        End If

        If leSupplier.Text = String.Empty Then
            MessageBox.Show("Please select the supplier of the item to be returned.")
            leSupplier.Focus()
            Exit Sub
        End If


        Dim TranBo As PDSATransaction = New PDSATransaction()
        Dim mgrbo As boManager = New boManager()

        Try
            mgrbo.DataObject.TransactionType = PDSATransactionType.Insert
            mgrbo.DataObject.Entity.amnt = 0
            mgrbo.DataObject.Entity.bodate = CDate(deBoDate.Text)
            mgrbo.DataObject.Entity.prevavlbl = 0
            mgrbo.DataObject.Entity.price = 0
            mgrbo.DataObject.Entity.qty = CInt(ceQty.Value)
            mgrbo.DataObject.Entity.sInsertid = PDSAAppConfig.CurrentLoginID
            mgrbo.DataObject.Entity.sLastUpdateid = PDSAAppConfig.CurrentLoginID
            mgrbo.DataObject.Entity.stckid = CInt(Trim(txtStckId.Text))
            mgrbo.DataObject.Entity.supcode = CInt(leSupplier.EditValue)
            TranBo.Add(mgrbo.DataObject)

            Dim mgrspBo As spBackOrderManager = New spBackOrderManager()
            mgrspBo.Entity.StckIid = CInt(txtStckId.Text)
            TranBo.Add(mgrspBo.DataObject)

            TranBo.Execute()
            DisableControls()
            MessageBox.Show("Back Order Entry Successfully Saved", "Saved")
            InitControls()
        Catch ex As PDSAValidationException
            MessageBox.Show(ex.Message)
            Exit Sub
        Catch ex As Exception
            MessageBox.Show(ex.ToString())
            Exit Sub
        End Try

    End Sub

    Private Sub EnableControls()
        btnSave.Enabled = True
        btnCancel.Enabled = True
        GroupControl1.Enabled = True
        btnClose.Enabled = False
        btnNew.Enabled = False
        deBoDate.Focus()
    End Sub

    Private Sub DisableControls()
        btnSave.Enabled = False
        btnCancel.Enabled = False
        GroupControl1.Enabled = False
        btnClose.Enabled = True
        btnNew.Enabled = True
        btnNew.Focus()
    End Sub

    Private Sub InitControls()
        deBoDate.EditValue = DateAndTime.Today
        txtItem.Text = ""
        txtStckId.Text = ""
        txtPrice.Text = ""
        ceQty.Value = 0
    End Sub

    Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        DisableControls()
        InitControls()
    End Sub

    Private Sub btnNew_Click(sender As Object, e As EventArgs) Handles btnNew.Click
        EnableControls()
        InitControls()
    End Sub

    Private Sub ceQty_LostFocus(sender As Object, e As EventArgs) Handles ceQty.LostFocus
        btnSave.Focus()
    End Sub
End Class
